File: 타겟넘버.py
global cnt
cnt = 0
def bfs(idx, value, numbers, target):
    if idx == len(numbers):
        if value == target:
            global cnt 
            cnt+=1
        return
    bfs(idx+1, value+numbers[idx],numbers,target)
    bfs(idx+1, value-numbers[idx], numbers, target)
def solution(numbers, target):
    answer = 0
    bfs(0,0,numbers, target)
    answer = cnt
    return answer

File: 타겟 넘버.java
class Solution {
    static int cnt =0;
    static public void dfs(int idx, int value, int[] numbers, int target){
        if(idx == numbers.length){
            if (value == target){
                cnt++;
            }
            return;
        }
        dfs(idx+1, value+numbers[idx], numbers, target);
        dfs(idx+1, value-numbers[idx], numbers, target);
    }
    public int solution(int[] numbers, int target) {
        int answer = 0;
        dfs(0,0, numbers, target);
        answer = cnt;
        return answer;
    }
}
